specification/consumption/resource-manager/Microsoft.Consumption/preview/2019-04-01-preview/consumption.json

```diff
@@ -0,0 +1,3146 @@
+          "BillingAccountUsageDetailsListForBillingPeriod": {
+            "$ref": "./examples/UsageDetailsListForBillingPeriodByBillingAccount.json"
+          },
+          "DepartmentUsageDetailsList": {
+            "$ref": "./examples/UsageDetailsListByDepartment.json"
+          },
+          "DepartmentUsageDetailsListForBillingPeriod": {
+            "$ref": "./examples/UsageDetailsListForBillingPeriodByDepartment.json"
+          },
+          "EnrollmentAccountUsageDetailsList": {
+            "$ref": "./examples/UsageDetailsListByEnrollmentAccount.json"
+          },
+          "EnrollmentAccountUsageDetailsListForBillingPeriod": {
+            "$ref": "./examples/UsageDetailsListForBillingPeriodByEnrollmentAccount.json"
+          },
+          "ManagementGroupUsageDetailsList": {
+            "$ref": "./examples/UsageDetailsListByManagementGroup.json"
+          },
+          "ManagementGroupUsageDetailsListForBillingPeriod": {
+            "$ref": "./examples/UsageDetailsListForBillingPeriodByManagementGroup.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/scopeUsageDetailsParameter"
+          },
+          {
+            "name": "$expand",
+            "description": "May be used to expand the properties/additionalInfo or properties/meterDetails within a list of usage details. By default, these fields are not included when listing usage details.",
+            "in": "query",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "$filter",
+            "description": "May be used to filter usageDetails by properties/resourceGroup, properties/resourceName, properties/resourceId, properties/chargeType, properties/reservationId or tags. The filter supports 'eq', 'lt', 'gt', 'le', 'ge', and 'and'. It does not currently support 'ne', 'or', or 'not'. Tag filter is a key value pair string where key and value is separated by a colon (:).",
+            "in": "query",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "$skiptoken",
+            "description": "Skiptoken is only used if a previous operation returned a partial result. If a previous response contains a nextLink element, the value of the nextLink element will include a skiptoken parameter that specifies a starting point to use for subsequent calls.",
+            "in": "query",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "$top",
+            "description": "May be used to limit the number of results to the most recent N usageDetails.",
+            "in": "query",
+            "required": false,
+            "type": "integer",
+            "minimum": 1,
+            "maximum": 1000
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/metricParameter"
```

Comment:
Is this the replacement for $apply?

Reply:
its not replacement. We are introducing a new Query Param from this version onwards.

Reply:
This changes the underlying metric we use to get data. The user has to pick which metric they want and get that data set. There isn't an option to get multiple, so it's not a $filter. (We thought about that and tried to start there, but it forced us to add more data and support a scenario we don't actually want to support, increasing surface area and introducing perf/scale risks due to the size of the request.)'